### ISSUE TITLE
Integrity check to report BibTex keys that do not fit the expected scheme

### ISSUE THREAD

**Issue by stefan-kolb** [JabRef/jabref#2206]
Example: 

Custom types (or even other types?) might miss the author field which leads to a not well-formed BibTeX key if no exception schema is defined.
We should probably warn in case something like this happens or at least provide an integrity check for this?

```
-@Standard{iso2011,
+@Standard{2011,
   title       = {{Systems and software engineering -- System and software Quality Requirements and Evaluation (SQuaRE) -- System and software quality models}},
   institution = {ISO/IEC},
   number      = {25010},
   year        = {2011},
   owner       = {Administrator},
@@ -563,11 +543,11 @@
   publisher = {Springer},
   year      = {2013},
   doi       = {10.1007/978-3-642-36981-0_75},
 }
```

**Comment by simonharrer** [comment]
I think a check to determine that the current bibtex key deviates from the one that would be generated is also helpful to reveal the entries which have been imported but deviate from the set generation scheme. 

Of course, in a save action, the key could always be generated, but I am not sure if this is advisable.

**Comment by matthiasgeiger** [comment]
Implemented in #2328.
